core/domain/classifier_services.py

```diff
@@ -305,47 +313,68 @@ def _update_classifier_training_job(classifier_training_job_model, status):
+                                   exp_version, state_name, training_data,
+                                   status):
+    """Creates a ClassifierTrainingJobModel in data store.
 
     Args:
         algorithm_id: str. ID of the algorithm used to generate the model.
+        interaction_id: str. ID of the interaction to which the algorithm
+            belongs.
         exp_id: str. ID of the exploration.
         exp_version: int. The exploration version at the time
             this training job was created.
         state_name: str. The name of the state to which the classifier
             belongs.
-        status: str. The status of the training job (NEW by default).
         training_data: dict. The data used in training phase.
-        job_id: str. The optional job_id which decides to create/update
-            classifier.
+        status: str. The status of the training job (NEW by default).
 
     Returns:
         job_id: str. ID of the classifier training job.
     """
-    classifier_training_job_model = (
-        classifier_models.ClassifierTrainingJobModel.get(job_id, False))
-    if classifier_training_job_model is None:
-        classifier_training_job = classifier_domain.ClassifierTrainingJob(
-            'job_id_dummy', algorithm_id, exp_id, exp_version,
-            state_name, status, training_data)
-        classifier_training_job.validate()
-        job_id = _create_classifier_training_job(classifier_training_job)
-    else:
-        classifier_training_job = get_classifier_training_job_from_model(
-            classifier_training_job_model)
-        classifier_training_job.validate()
-        _update_classifier_training_job(classifier_training_job_model,
-                                        status)
+    classifier_training_job = classifier_domain.ClassifierTrainingJob(
+        'job_id_dummy', algorithm_id, interaction_id, exp_id, exp_version,
+        state_name, status, training_data)
+    classifier_training_job.validate()
+    job_id = _create_classifier_training_job(classifier_training_job)
     return job_id
 
 
+def update_classifier_training_job(classifier_training_job):
+    """Checks for the existence of the model and then updates it.
+
+    Args:
+        classifier_training_job: ClassifierTrainingJob. Domain object for the
+            classifier training job model.
+
+    Raises:
+        Exception. The ClassifierTrainingJobModel corresponding to the job_id
+            of the ClassifierTrainingJob does not exist.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(
+            classifier_training_job.job_id, strict=False))
+    if not classifier_training_job_model:
+        raise Exception(
+            'The ClassifierTrainingJobModel corresponding to the job_id of the'
+            'ClassifierTrainingJob does not exist.')
+    classifier_training_job.validate()
+    _update_classifier_training_job(classifier_training_job_model,
```

Comment:
Could you just inline this? Not sure you need to break it out into a separate function.